packages/guideline-blocks-settings/src/settings/marginExtended.ts

```diff
@@ -0,0 +1,91 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import {
+    MultiInputLayout,
+    SettingBlock,
+    appendUnit,
+    maximumNumericalOrPixelOrAutoRule,
+    numericalOrPixelRule,
+    presetCustomValue,
+} from '../';
+import { MARGIN_DEFAULT_PLACEHOLDER } from './defaultValues';
+import { getMarginSlider } from './margin';
+import { marginStyleMap } from './types';
+
+type MarginSettingsType = {
+    id?: string;
+};
+
+/**
+ * Returns margin settings: margin switch, margin slider, custom margin input for every direction
+ *
+ * @param options Options for the settings
+ * @param options.id Custom suffix for the setting ids
+ * @returns {SettingBlock} Returns margin settings
+ */
+export const getMarginExtendedSettings = (options?: MarginSettingsType): SettingBlock => {
```

Comment:
None except one settings helper has tests, please add some

Reply:
Added tests to all other setting helpers as well.